packages/vite/src/node/config.ts

```diff
@@ -383,9 +383,20 @@ export async function resolveConfig(
   }
 
   // resolve root
-  const resolvedRoot = normalizePath(
-    config.root ? path.resolve(config.root) : process.cwd()
-  )
+  let resolvedRoot: string
+  if (!config.root) {
+    resolvedRoot = process.cwd()
+  } else if (path.isAbsolute(config.root)) {
+    resolvedRoot = config.root
+  } else {
+    if (typeof configFile !== 'string') {
+      throw new Error(
+        `"config.root" cannot be relative when config file is not used.`
+      )
+    }
+    resolvedRoot = path.resolve(path.dirname(configFile), config.root)
```

Comment:
> Can be an absolute path, or a path relative to the location of the config file itself.
> https://github.com/vitejs/vite/blob/3d14372c8fa1acb426923c439e545211fc9d7ccd/docs/config/shared-options.md

According to the docs, it is relative to **the location of the config file**. So accurately it should be without `path.dirname`.
But I think this is more intuitive. For example, when config is located at `/foo/bar/vite.config.js` and wants to locate root at `/foo/bar`, `config.root` needs to be `../` if `path.dirname` is removed.

Should I change the docs to some other words? or should I remove `path.dirname`?

Reply:
I think it wasn't properly worded, and "the location of the config file" implies `path.dirname` as you did here.